packages/keystone/databaseAdapters/utils.js

```diff
@@ -63,9 +67,38 @@ function matchDatabase (mapping, str) {
     return undefined
 }
 
+function getDatabaseAdapter (keystone, list = '*') {
+    const databaseUrl = get(conf, 'DATABASE_URL')
+
+    if (databaseUrl.startsWith('postgres')) {
+        return keystone.adapter
+    } else if (databaseUrl.startsWith('custom')) {
+        if (list === '*') {
+            return keystone.adapter.__databaseAdapters.default
+        }
+        throw new Error('Multiple list adapters support is not implemented!')
+    } else {
+        throw new Error('invalid database adapter')
+    }
+}
+
+function getListAdapters (keystone) {
+    const databaseUrl = get(conf, 'DATABASE_URL')
+
+    if (databaseUrl.startsWith('postgres')) {
+        return keystone.adapter.listAdapters
+    } else if (databaseUrl.startsWith('custom')) {
```

Comment:
What's the custom db url?

Reply:
I need to add an example of service configuration. Custom url is a format that was introduced for `ScalableDatabaseAdapter`

Reply:
In fact, this is just an object, that says app where to push data to write and where to read it